File: result/translate/starchat/java2python/python/CHECK_POSSIBLE_TRANSFORM_ONE_STRING_ANOTHER.py
def f_gold(s1, s2):
        n = len(s1)
        m = len(s2)
        dp = [[False] * (m + 1) for _ in range(n + 1)]
        for i in range(n + 1):
            for j in range(m + 1):
                if i == 0 and j == 0:
                    dp[i][j] = True
                elif i == 0:
                    dp[i][j] = dp[i][j - 1] and (s2[j - 1].isupper() or s1[i - 1] == s2[j - 1])
                elif j == 0:
                    dp[i][j] = dp[i - 1][j] and (s1[i - 1].isupper() or s1[i - 1] == s2[j - 1])
                else:
                    dp[i][j] = dp[i - 1][j] and (s1[i - 1] == s2[j - 1]) or (dp[i][j - 1] and (s

File: result/translate/starchat/java2cpp/c++/CHECK_POSSIBLE_TRANSFORM_ONE_STRING_ANOTHER.cpp
bool f_gold ( string s1, string s2 ) {
        int n = s1. length ( );
        int m = s2. length ( );
        vector<vector<bool>> dp (n + 1, vector<bool> (m + 1, false));
        for ( int i = 0; i <= n; i++ ) dp[i][0] = true;
        for ( int j = 0; j <= m; j++ ) dp[0][j] = false;
        for ( int i = 1; i <= n; i++ ) {
            for ( int j = 1; j <= m; j++ ) {
                if ( s1[i - 1] == s2[j - 1] ||!isupper(s1[i - 1]) ) dp[i][j] = dp[i - 1][j - 1];
                if ( dp[i][j] &&!isupper(s1[i - 1]) ) dp[i][j] = true;
            }
        }
        return dp[n][m];
    }